source/navier_stokes_preconditioner.cc

```diff
@@ -1968,6 +1988,13 @@ NavierStokesPreconditioner<dim>::local_assemble_preconditioner(
                                     0. :
                                     integration_helper.values_unit_cell[i][q] *
                                       time_stepping.weight();
+
+                                // Note: The division by weight_rho is necessary since
+                                // in the assignment of data_for_val[i][q] it is
+                                // multiplied by weight_rho.
+                                phi_and_grad -= weight_damping / weight_rho *
+                                                integration_helper.values_unit_cell[i][q];
```

Comment:
This reminds me of this ugly setup with repeating the code with matrices in both cases.